sdk/sdktypes/user.go

```
@@ -0,0 +1,37 @@
+package sdktypes
+
+import (
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+
+	userv1 "go.autokitteh.dev/autokitteh/proto/gen/go/autokitteh/users/v1"
+)
+
+type User struct{ object[*UserPB, UserTraits] }
+
+var InvalidUser User
+
+type UserPB = userv1.User
+
+type UserTraits struct{}
+
+func (UserTraits) Validate(m *UserPB) error       { return nil }
+func (UserTraits) StrictValidate(m *UserPB) error { return nil }
+
+func UserFromProto(m *UserPB) (User, error) { return FromProto[User](m) }
+
+func (u User) Data() map[string]string { return u.read().Data }
+func (u User) Provider() string        { return u.read().Provider }
+
+func NewUser(provider string, data map[string]string) User {
+	return kittehs.Must1(UserFromProto(&UserPB{
+		Provider: provider,
+		Data:     data,
+	}))
+}
+
+func (u User) UniqueID() (id string) {
+	if id = u.Data()["email"]; id == "" {
```

are you sure we can count on email here?
if email doesn't exists, all users without email would get the same <unknown> id, won't that be a problem?

---

"id", it's more of a unique qualifier to display. maybe i should change it to "title". I've seen everywhere email set.

---

so maybe we should panic or error if email is not set ?
I worry about case where email is not set for multiple users and they get the same <unknown> id

---

it's for display only - we're not using it for anything else. the function name is wrong - will change it to `Title`.